File: T1.SqlSharp/T1.SqlSharp/Expressions/SqlOrderColumn.cs
namespace T1.SqlSharp.Expressions;

public class SqlOrderColumn : ISqlExpression
{
    public SqlType SqlType { get; } = SqlType.OrderColumn;
    public TextSpan Span { get; set; } = new();
    public void Accept(SqlVisitor visitor)
    {
        visitor.Visit_OrderColumn(this);
    }

    public required ISqlExpression ColumnName { get; set; }
    public OrderType Order { get; set; }

    public string ToSql()
    {
        return $"{ColumnName.ToSql()} {Order.ToString().ToUpper()}";
    }
}

File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/SqlOrderColumn.ts
import { SqlType } from './SqlType';
import { TextSpan } from '../StringParser';
import { ISqlExpression, SqlVisitor } from './ISqlExpression';
import { OrderType } from './OrderType';

export class SqlOrderColumn implements ISqlExpression {
    SqlType: SqlType = SqlType.OrderColumn;
    Span: TextSpan = new TextSpan();
    ColumnName!: ISqlExpression;
    Order!: OrderType;

    Accept(visitor: SqlVisitor): void {
        visitor.Visit_OrderColumn(this);
    }

    ToSql(): string {
        return `${this.ColumnName.ToSql()} ${this.Order.toString().toUpperCase()}`;
    }
} 